example/src/CrudWebserverDb.hs

```diff
@@ -0,0 +1,415 @@
+         <*> v .: "age"
+
+instance ToJSON User where
+  toJSON (User name age) =
+    object [ "name" .= name
+           , "age"  .= age
+           ]
+
+-- For testing we will also need to generate arbitrary users.
+
+instance Arbitrary User where
+  arbitrary = User <$> (T.pack <$> (arbitrary `suchThat` all isPrint))
+                   <*> arbitrary
+
+-- Note that there's a constraint on the username that only allows
+-- printable characters. We'll come back to why this is later.
+
+------------------------------------------------------------------------
+
+-- Servant let's us describe the API of the webserver as a datatype.
+
+-- Our webserver will have an endpoint for creating a user, which takes
+-- a JSON-encoded @User@ in the body of an POST request and returns a
+-- unique id. The rest of the endpoints capture said id in the URL and
+-- get the user associated with it, increment the age of the user, and
+-- delete the user respectively. Finally we also have a health endpoint
+-- which helps us determine if the webserver is up and running.
+
+type Api =
+       "user" :> "add" :> ReqBody '[JSON] User     :> Post '[JSON] (Key User)
+  :<|> "user" :> "get" :> Capture "key" (Key User) :> Get  '[JSON] (Maybe User)
+  :<|> "user" :> "inc" :> Capture "key" (Key User) :> Get  '[JSON] ()
+  :<|> "user" :> "del" :> Capture "key" (Key User) :> Get  '[JSON] ()
+  :<|> "health"                                    :> Get  '[JSON] ()
+
+api :: Proxy Api
+api = Proxy
+
+------------------------------------------------------------------------
+
+-- Next we describe how the endpoints are implemented.
+
+server :: ConnectionPool -> Server Api
+server pool =
+  userAdd :<|> userGet :<|> userBirthday :<|> userDelete :<|> health
+  where
+  userAdd :: MonadIO io => User -> io (Key User)
+  userAdd newUser = liftIO $ runSqlPersistMPool (insert newUser) pool
+
+  userGet :: MonadIO io => Key User -> io (Maybe User)
+  userGet key = liftIO $ runSqlPersistMPool (get key) pool
+
+  userBirthday :: MonadIO io => Key User -> io ()
+  userBirthday key = liftIO $
+    runSqlPersistMPool (update key [UserAge +=. 1]) pool
+
+  userDelete :: MonadIO io => Key User -> io ()
+  userDelete key = liftIO $ runSqlPersistMPool (delete key) pool
+
+  health :: MonadIO io => io ()
+  health = return ()
+
+app :: ConnectionPool -> Application
+app pool = serve api $ server pool
+
+mkApp :: FilePath -> IO Application
+mkApp sqliteFile = do
+  pool <- runNoLoggingT $ do
+    createSqlitePool (cs sqliteFile) 1
+
+  runSqlPool (runMigration migrateAll) pool
+  return $ app pool
+
+runServer :: FilePath -> Warp.Port -> IO ()
+runServer sqliteFile port = Warp.run port =<< mkApp sqliteFile
+
+------------------------------------------------------------------------
+
+-- We get the client bindings for free from the API datatype.
+
+postUserC    :: User     -> ClientM (Key User)
+getUserC     :: Key User -> ClientM (Maybe User)
+putUserC     :: Key User -> ClientM ()
```

Comment:
`putUserC`? Seems like a strange name, shouldn't it be something like `incUserC` or even `incAgeUserC`.